transmart-core-db-tests/src/integration-test/groovy/org/transmartproject/db/multidimquery/HibernateCriteriaQueryBuilderSpec.groovy

```diff
@@ -443,4 +448,54 @@ class HibernateCriteriaQueryBuilderSpec extends Specification {
         results.size() == expectedResults.size()
         results.sort() == expectedResults.sort()
     }
+
+    void 'test queries for empty database'() {
+        given: 'No studies loaded'
+        TestData.clearAllData()
+        QueryBuilder builder = HibernateCriteriaQueryBuilder.forStudies([])
+
+        when: 'Querying for all observations'
+        def criteria = builder.buildCriteria(new TrueConstraint())
+        def results = getList(criteria) as List
+
+        then: 'The result is empty'
+        results.empty
+    }
+
+    void 'test queries for user with no access to any study'() {
+        given: 'No access to any study'
+        setupHypercubeData()
+        QueryBuilder builder = HibernateCriteriaQueryBuilder.forStudies([])
```

Comment:
It seems to me like the previous test does not add anything new.
`HibernateCriteriaQueryBuilder` works with studies supplied via `forStudies(List)` method.
Both, no studies in the database or no studies user have access to conduct as an empty list of studies.